File: bdp-govern/quality/src/main/scala/org/apache/griffin/measure/configuration/enums/WriteMode.scala
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.apache.griffin.measure.configuration.enums

/**
 * write mode when write metrics and records
 */
sealed trait WriteMode {}

object WriteMode {
  def defaultMode(procType: ProcessType.ProcessType): WriteMode = {
    procType match {
      case ProcessType.BatchProcessType => SimpleMode
      case ProcessType.StreamingProcessType => TimestampMode
    }
  }
}

/**
 * simple mode: write metrics and records directly
 */
case object SimpleMode extends WriteMode {}

/**
 * timestamp mode: write metrics and records with timestamp information
 */
case object TimestampMode extends WriteMode {}

File: lake/arctic/spark/v3.1/spark/src/main/java/com/netease/arctic/spark/writer/WriteMode.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.netease.arctic.spark.writer;

public enum WriteMode {
  OVERWRITE_BY_FILTER("overwrite-by-filter"),
  OVERWRITE_DYNAMIC("overwrite-dynamic"),
  APPEND("append"),
  UPSERT("upsert"),
  MERGE("merge");

  public static final String WRITE_MODE_KEY = "write-mode";

  public final String mode;
  WriteMode(String mode) {
    this.mode = mode;
  }

  public static WriteMode getWriteMode(String mode) {
    for (WriteMode m : values()) {
      if (m.mode.equalsIgnoreCase(mode)) {
        return m;
      }
    }
    throw new IllegalArgumentException("Invalid write mode: " + mode);
  }
}
